src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -658,43 +658,61 @@ def remove_file_post
     end
   end
 
-  describe 'GET #package_live_build_log' do
-    context "successfully" do
-      before do
-        get :live_build_log, project: source_project, package: source_package, repository: '10.2', arch: 'i586', format: 'js'
+  context "build logs" do
+    RSpec.shared_examples "build log" do
+      context "successfully" do
+        before do
+          do_request project: source_project, package: source_package, repository: '10.2', arch: 'i586', format: 'js'
+        end
+
+        it { expect(response).to have_http_status(:ok) }
       end
 
-      it { expect(response).to have_http_status(:ok) }
-    end
+      context "with a protected package" do
+        let(:flag) { create(:sourceaccess_flag, project: source_project) }
```

Comment:
Why not 
``
let!(:flag) { create(:sourceaccess_flag, project: source_project) } 
``
and dropping line 675?